contributing.md

```diff
@@ -0,0 +1,131 @@
+
+### Testing the modifications
+
+It is easy to build the kernel, the dependences, and start a VM thanks to
+[MPTCP Upstream Virtme Docker](https://github.com/multipath-tcp/mptcp-upstream-virtme-docker).
+With just one command, you can get a prompt from a VM using the modified kernel
+or run the whole test suite. For more details, please see this
+[page](https://github.com/multipath-tcp/mptcp_net-next/wiki/CI) from our wiki.
+
+
+{: .note}
+Any push to a fork of the [`mptcp_net-next`](https://github.com/multipath-tcp/mptcp_net-next)
+repository on GitHub will trigger the CI: look at the `Actions` tab in the fork
+repository to follow the progress, and show the results.
+
+### Patches are shared on the mailing list
+
+#### Why?
+
+Like many kernel subsystems, patches can be sent to the assigned mailing list
+([mptcp@lists.linux.dev](mailto:mptcp@lists.linux.dev)), and **not** via Pull
+Requests on GitHub. The idea is not to restrict access to people who have a
+GitHub account, and to have a decentralized system.
+
+#### It is easy with `b4`
+
+Sending patches by emails is **not difficult**, not more than setting up a
+GitHub account, with the verification of the email address, a new SSH key, etc.
+
+To help with this task, we recommend to use
+[b4](https://b4.docs.kernel.org/en/latest/contributor/overview.html). After
+having [installed](https://b4.docs.kernel.org/en/latest/installing.html) it, you
+need to either:
+- configure the
+[web-endpoint](https://b4.docs.kernel.org/en/latest/contributor/send.html)
+(launching 3 commands)
+- or set info about your [SMTP server](https://git-send-email.io/#step-2) in the
+  git config file
+
+#### Workflow
+
+The workflow is then easy: use
+[`b4 prep`](https://b4.docs.kernel.org/en/latest/contributor/prep.html) to
+create a new branch, and
+[`b4 send`](https://b4.docs.kernel.org/en/latest/contributor/send.html) to send
+your patches by email.
+
+In short, once `b4` has been installed and configured, the workflow looks like:
+
+```bash
+git switch export # or export-net for fixes
+b4 prep -n <new branch name>
+# edit your code
+git add -p && git commit -s
+./scripts/checkpatch.pl -g HEAD ## check the output, fix issues with 'git commit --amend'
+b4 prep --edit-cover # if you have more than one patch
+b4 send --reflect # check if the mails you sent to yourself are OK
+b4 send
+```
+
+Once sent, the patches will be tested by our CI. A report will then be sent on
+[Patchwork](https://patchwork.kernel.org/project/mptcp/list/?state=*). See the
+[CI](https://github.com/multipath-tcp/mptcp_net-next/wiki/CI) page on our wiki
+for more details.
+
+### Commit messages
+
+Please **always** include the reason why a patch is needed: we can easily list
+the modifications by looking at the patch itself (`diff`), but understanding why
+a modification has been done, why done this way and not another, etc. is harder
+to find. This should then appear in the commit message.
```

Comment:
Instead of:

> is important context for a change that is important for reviewers and future maintenance.

Maybe just:

> is important context for reviewers and future maintenance.

no?

Reply:
> Instead of:
> 
> > is **important** context for a change that is **important** for reviewers and future maintenance.
> 
> Maybe just:
> 
> > is **important** context for reviewers and future maintenance.
> 
> no?

I suggest merging this like that for the moment (so I don't forget about this PR :) ). We can refine this text later if needed.